cpp/tests/io/parquet_test.cpp

```diff
@@ -4260,6 +4260,175 @@ TEST_F(ParquetWriterTest, CheckColumnOffsetIndexStruct)
+    "\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+    // in-range 2-byte UTF8 (U+00E9). should truncate to "éééê"
+    "ééééé",
+    // max 2-byte UTF8 (U+07FF). should not truncate
+    "߿߿߿߿߿",
+    // in-range 3-byte UTF8 (U+0800). should truncate to "ࠀࠁ"
+    "ࠀࠀࠀ",
+    // max 3-byte UTF8 (U+FFFF). should not truncate
+    "\xef\xbf\xbf\xef\xbf\xbf\xef\xbf\xbf",
+    // in-range 4-byte UTF8 (U+10000). should truncate to "𐀀𐀁"
+    "𐀀𐀀𐀀",
+    // max unicode (U+10FFFF). should truncate to \xf4\x8f\xbf\xbf\xf4\x90\x80\x80,
+    // which is no longer valid unicode, but is still ok UTF-8???
+    "\xf4\x8f\xbf\xbf\xf4\x8f\xbf\xbf\xf4\x8f\xbf\xbf",
+    // max 4-byte UTF8 (U+1FFFFF). should not truncate
+    "\xf7\xbf\xbf\xbf\xf7\xbf\xbf\xbf\xf7\xbf\xbf\xbf"};
+
+  // NOTE: UTF8 min is initialized with 0xf7bfbfbf. Binary values larger
+  // than that will not become minimum value (when written as UTF-8).
+  const char* truncated_min[] = {"yyyyyyyy",
+                                 "\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x7f",
+                                 "\xf7\xbf\xbf\xbf",
+                                 "éééé",
+                                 "߿߿߿߿",
+                                 "ࠀࠀ",
+                                 "\xef\xbf\xbf\xef\xbf\xbf",
+                                 "𐀀𐀀",
+                                 "\xf4\x8f\xbf\xbf\xf4\x8f\xbf\xbf",
+                                 "\xf7\xbf\xbf\xbf"};
+
+  const char* truncated_max[] = {"yyyyyyyz",
+                                 "\x7f\x7f\x7f\x7f\x7f\x7f\x7f\x80",
+                                 "\xff\xff\xff\xff\xff\xff\xff\xff\xff",
+                                 "éééê",
+                                 "߿߿߿߿߿",
+                                 "ࠀࠁ",
+                                 "\xef\xbf\xbf\xef\xbf\xbf\xef\xbf\xbf",
+                                 "𐀀𐀁",
+                                 "\xf4\x8f\xbf\xbf\xf4\x90\x80\x80",
+                                 "\xf7\xbf\xbf\xbf\xf7\xbf\xbf\xbf\xf7\xbf\xbf\xbf"};
+
+  column_wrapper<cudf::string_view> col0{coldata[0]};
+  column_wrapper<cudf::string_view> col1{coldata[1]};
+  column_wrapper<cudf::string_view> col2{coldata[2]};
+  column_wrapper<cudf::string_view> col3{coldata[3]};
+  column_wrapper<cudf::string_view> col4{coldata[4]};
+  column_wrapper<cudf::string_view> col5{coldata[5]};
+  column_wrapper<cudf::string_view> col6{coldata[6]};
+  column_wrapper<cudf::string_view> col7{coldata[7]};
+  column_wrapper<cudf::string_view> col8{coldata[8]};
+  column_wrapper<cudf::string_view> col9{coldata[9]};
+
+  std::vector<std::unique_ptr<column>> cols;
+  cols.push_back(col0.release());
+  cols.push_back(col1.release());
+  cols.push_back(col2.release());
+  cols.push_back(col3.release());
+  cols.push_back(col4.release());
+  cols.push_back(col5.release());
+  cols.push_back(col6.release());
+  cols.push_back(col7.release());
+  cols.push_back(col8.release());
+  cols.push_back(col9.release());
```

Comment:
```suggestion
auto const cols = [&] {
    std::vector<string_view> views;
    for(int i = 0; ....) { views[i] = string_view{coldata[i]}.release(); }
    return views;
}();

```

Reply:
I'll switch this to how @hyperbolic2346 has been doing the tests lately.  Hopefully that will look better.

Reply:
Couldn't figure out how to get a valid vector of `column_view`s out of a lambda to go straight to a table_view for this one, so stuck with `column` pointers as suggested.

Reply:
You can't get a `column_view` without storing a column.